File: 3903-smallest-stable-index-i/3903-smallest-stable-index-i.c
#define max(a , b) ((a) > (b) ? (a) : (b))
#define min(a , b) ((a) < (b) ? (a) : (b))

int firstStableIndex(int* nums, int numsSize, int k) {
    if(numsSize <= 1){
        return 0;
    }

    
    for(int i = 0; i < numsSize; i++){
        int leftmax = nums[i];
        int rightmin = nums[i];
        
        for(int k = 0; k < i; k++){
            leftmax = max(leftmax , nums[k]);
        }

        for(int j = i + 1; j < numsSize; j++){
            rightmin = min(rightmin , nums[j]);
        }

        if(leftmax - rightmin <= k){
            return i;
        }
    }
    return -1;
}

File: 3903-smallest-stable-index-i/3903-smallest-stable-index-i.py
class Solution:
    def firstStableIndex(self, nums: list[int], k: int) -> int:
        if len(nums) <= 1:
            return 0
        
        for i in range(len(nums)):
            leftmax = nums[i]
            rightmin = nums[i]

            for j in range(i):
                leftmax = max(leftmax , nums[j])
            for m in range(i + 1 , len(nums)):
                rightmin = min(rightmin , nums[m])
            if leftmax - rightmin <= k:
                return i
        return -1